lib/entry.js

```diff
@@ -3,14 +3,14 @@
 var DIRECTORY_MODE = 16877;
 
 module.exports = Entry;
-function Entry(relativePath, size, mtime) {
+function Entry(relativePath, size, mtime, mode) {
   var isDirectory = relativePath.charAt(relativePath.length - 1) === '/';
 
    // ----------------------------------------------------------------------
    // required properties
 
   this.relativePath = relativePath;
-  this.mode = isDirectory ? DIRECTORY_MODE : 0;
+  this.mode = mode || (isDirectory ? DIRECTORY_MODE : 0);
```

Comment:
I'm +1 on using `undefined` as well. I feel like it makes it easier to add to/modify the signature later if needed. Though I don't feel too strongly about it.